### ISSUE TITLE
Is tf2onnx support BatchNormalization layer from tensorflow to onnx ?

### ISSUE THREAD

**Issue by Center0710** [onnx/tensorflow-onnx#1078]
**Problem Description**

Is tf2onnx support BatchNormalization layer from tensorflow to onnx  ? I can see support from the document in the following URL, But I am having trouble during the conversion

[https://github.com/onnx/onnx/blob/master/docs/Operators.md](url)

**Describe the bug**

 I use tensorflow to build a simple network( conv2D + BatchNormalization +leakyRelu), the network structure is as shown below

![1](https://user-images.githubusercontent.com/56102982/91421537-a0408280-e888-11ea-990e-141bd57d3059.PNG)

And then, PB model ' s structure is like this
![2](https://user-images.githubusercontent.com/56102982/91421692-cf56f400-e888-11ea-87fc-f4695f90671a.PNG)

I use tf2onnx  convert PB model to onnx model, but onnx model is like this
![3](https://user-images.githubusercontent.com/56102982/91421945-1e048e00-e889-11ea-9c40-df43a1c40b48.PNG)

Finally, I use onnxruntime to inference my onnx model, but get many nan value

**System information**

- OS : Win10
- Tensorflow version : 1.14
- Python version : 3.6.2
- tf2onnx version : 1.6.3
- command line : 
python -m tf2onnx.convert --input model.pb --output model.onnx --inputs Placeholder_1:0 --outputs conv2D/Sigmoid:0 --opset 11

**To Reproduce**
I can share this pb model and more info

**Comment by TomWildenhain-Microsoft** [comment]
The Batch Normalization op is supported and should work.  We'll need to look into this.

**Comment by Center0710** [comment]
**Update question**

- **Old question**：

Sorry, this is my problem. I noticed that the BatchNormalization of my pb model is a bit strange, so I searched the tensorflow code and found the following error. This is the training/inference switch set by my batchnormalization. After modifying it, the BatchNormalization of the pb model is normal.

  ![4](https://user-images.githubusercontent.com/56102982/91513463-fa882480-e916-11ea-95cb-fc9b92d92f69.PNG)

- **New question**：

After the above modification, the structure of my pb model looks like this

  ![5](https://user-images.githubusercontent.com/56102982/91513862-f4df0e80-e917-11ea-9e0b-3a8bcb58bbfe.PNG)

And then, my onnx model looks like this, **BatchNormalization layer disappeared**

  ![6](https://user-images.githubusercontent.com/56102982/91514217-c6adfe80-e918-11ea-88ed-c43b27b4944b.PNG)

I checked the information about the conversion process and found  **BatchNormalization -1 (1->0)**

  ![7](https://user-images.githubusercontent.com/56102982/91514351-1ee50080-e919-11ea-8501-2a270ee02895.PNG)

- **summary of the issue**：

 when I convert ckpt to pb file, **FusedBatchNorm layer** can be generated smoothly, but when I use tf2onnx to convert pb to onnx, the **BatchNormalization layer will disappear automatically**

**System information**

As at the beginning of the article
 
**To Reproduce**

The model download link is as follows

https://drive.google.com/file/d/1egEGJmxLmGvvlxCg57ApAGVPMg4CoyQj/view

**Comment by phager90** [comment]
@Center0710: BN layers are mainly important for training. 
For inference these layers can be folded into the neighboring convolution layer to speed up the inference time.

See here for a more detailed explanation:
https://towardsdatascience.com/speed-up-inference-with-batch-normalization-folding-8a45a83a89d8

So your BN layer has not disappeared, it has been integrated into the conv layer.

**Comment by guschmue** [comment]
Yes, whenever possible we fuse the bn into conv:
https://github.com/onnx/tensorflow-onnx/blob/master/tf2onnx/optimizer/back_to_back_optimizer.py#L182

**Comment by Center0710** [comment]
@TomWildenhain-Microsoft @phager90 @guschmue 

Thank you for answering, the problem has been solved
